src/igd_async.rs

```diff
@@ -67,11 +67,47 @@ impl Gateway {
             tx.send(res)
         });
 
-        rx.map_err(|_| panic!("worker thread didn't send the result?"))
+        rx.map_err(|_| {
+                // Note: this will always fail.
+                // the fact that we reached this line of code means that `send` was dropped without
+                // sending anything. The type here is used to enforce the fact that the
+                // worker thread tried to call `.send()`.
+                let res: Result<(), Result<SocketAddrV4, GetAnyAddressError>> = unwrap!(jh.join());
```

Comment:
Isn't using `unwrap!` the same as `panic!` in means of terminating execution?

Reply:
Yes, that line of code will always panic. It's not possible for use to have reached and the worker thread to not have panicked, so we're just propagating the panic.

Reply:
Hm, why not to map the failure into error and continue with possibility to recover and retry?